internal/util/kongversion.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"sync"
+
+	"github.com/blang/semver/v4"
+)
+
+var (
+	kongVersion     = semver.MustParse("0.0.0")
+	kongVersionOnce sync.Once
+)
+
+// SetKongVersion sets the Kong version. It can only be used once. Repeated calls will not update the Kong
+// version
+func SetKongVersion(version semver.Version) {
+	kongVersionOnce.Do(func() {
+		kongVersion = version
+	})
+}
+
+// GetKongVersion retrieves the Kong version. If the version is not set, it returns the lowest possible version
+func GetKongVersion() semver.Version {
+	return kongVersion
+}
```

Comment:
It seems to me that knowing the version of the gateway is something that might be good to live in `dataplane.KongClient` as `c.GetKongGatewayVersion` and then `SetVersion` doesn't need to exist because the underlying `kongConfig` client can already determine this? What are you thoughts? :thinking:

Reply:
I believe it actually does include it already (it was used for the original mTLS gate and I didn't rip it out entirely). The intent here was to provide a module-scope version so that we don't care if you have access to the client, which may be the case either in helper functions that don't have a client receiver or in code outside the parser (controllers, admission webhook, etc.) that wouldn't have access to the client normally.